nodebuilder/state/module.go

```diff
@@ -26,7 +30,14 @@ func ConstructModule(tp node.Type, cfg *Config, coreCfg *core.Config) fx.Option
 		fx.Supply(*cfg),
 		fx.Supply(cfg.GranterAddress),
 		fx.Error(cfgErr),
-		fxutil.ProvideIf(coreCfg.IsEndpointConfigured(), fx.Annotate(coreAccessor,
+		fxutil.ProvideIf(coreCfg.IsEndpointConfigured(), fx.Annotate(
```

Comment:
Fx cleanliness nit: you can also just provide a state.Option in case cfg.GranterAddr != nil and take []state.Option as param in module's core access constructor